api-inference-community/docker_images/pyannote/app/pipelines/automatic_speech_recognition.py

```diff
@@ -0,0 +1,38 @@
+import datetime
+from typing import Dict
+
+import numpy as np
+import torch
+from app.pipelines import Pipeline
+from pyannote.audio import Pipeline as Pypeline
+
+
+class AutomaticSpeechRecognitionPipeline(Pipeline):
+    def __init__(self, model_id: str):
+        # IMPLEMENT_THIS
+        # Preload all the elements you are going to need at inference.
+        # For instance your model, processors, tokenizer that might be needed.
+        # This function is only called once, so do all the heavy processing I/O here
+        # IMPLEMENT_THIS : Please define a `self.sampling_rate` for this pipeline
+        # to automatically read the input correctly
+        self.sampling_rate = 16000
```

Comment:
I see the Pypeline is able to resample on its own.

Is there a way to get the real value of the expected sampling rate from `Pypeline` ?
It will just avoid an unecessary resampling, since the inputs will be sent to you directly correctly sampled.

Reply:
There is no easy way to query this...
FWIW, all existing models (for now) have been trained with `sample_rate = 16000`.

Reply:
Let's stick with this then.